tests/fail/data_race/dealloc_read_race1.rs

```diff
@@ -19,11 +19,16 @@ pub fn main() {
 
     unsafe {
         let j1 = spawn(move || {
-            *ptr.0
+            let _val = *ptr.0;
         });
```

Comment:
This is a behavior change, but I did this to preserve the way both `spawn` use a multiline block. This is similar to https://github.com/rust-lang/miri/pull/2244#discussion_r901285805. With the behavior change it still correctly tests what the test was intended to cover.